rust-toolchain.toml

```diff
@@ -2,4 +2,4 @@
 # The default profile includes rustc, rust-std, cargo, rust-docs, rustfmt and clippy.
 # https://rust-lang.github.io/rustup/concepts/profiles.html
 profile = "default"
-channel = "1.75.0"
+channel = "1.76.0"
```

Comment:
I think clippy is failing because the toolchain is updating, so a newer clippy is running. I'll take a look tomorrow at fixing what it's complaining about. 

Alternatively if this is left at 1.75 and CI passes I'll just merge it 👍

Reply:
Thank you! I don't get it, according to https://rust-lang.github.io/rust-clippy/master/index.html#/redundant_guards, redundant_guards were introduced in 1.73.0

But, let's see what you find.

Reply:
@sstadick Any news?
Thanks!

Reply:
Thanks for the bump! I have no idea why this hasn't been failing earlier. It seems like it should have been failing for a while, and looking at previous CI pipelines it's been using > 1.73 🤷 

The suggestion it's making is correct, would you mind applying it in this PR? Or I can make a PR into yours. Whatever works for you.

Reply:
This PR should fix it, and includes your changes: https://github.com/sstadick/hck/pull/86

Reply:
Thanks @sstadick 

🤦‍♂️ I must have mixed **m** and **n** because, I've tried what I thought it was this change locally.

Like you, I was surprised this wasn't failing previously but, I'm happy it's fixed.

On a general note, would you mind if I keep doing this every 2 - 3 months?

Reply:
I would not mind at all, and would greatly appreciate it! I'll try to be a bit speedier getting things merged in 👍